apps/webservice/src/app/[workspaceSlug]/systems/SystemActionsDropdown.tsx

```diff
@@ -1,150 +1,53 @@
-      .then(() => {
-        utils.system.list.invalidate({
-          workspaceId: system.workspaceId,
-        });
-        onSubmit?.();
-        setOpen(false);
-      }),
-  );
-
-  return (
-    <Dialog open={open} onOpenChange={setOpen}>
-      <DialogTrigger asChild>{children}</DialogTrigger>
-      <DialogContent onClick={(e) => e.stopPropagation()}>
-        <DialogHeader>
-          <DialogTitle>Edit System</DialogTitle>
-        </DialogHeader>
-
-        <Form {...form}>
-          <form onSubmit={onFormSubmit} className="space-y-3">
-            <FormField
-              control={form.control}
-              name="name"
-              render={({ field }) => (
-                <FormItem>
-                  <FormLabel>Name</FormLabel>
-                  <FormControl>
-                    <Input {...field} />
-                  </FormControl>
-                </FormItem>
-              )}
-            />
-            <FormField
-              control={form.control}
-              name="slug"
-              render={({ field }) => (
-                <FormItem>
-                  <FormLabel>Slug</FormLabel>
-                  <FormControl>
-                    <Input {...field} />
-                  </FormControl>
-                </FormItem>
-              )}
-            />
-            <FormField
-              control={form.control}
-              name="description"
-              render={({ field }) => (
-                <FormItem>
-                  <FormLabel>Description</FormLabel>
-                  <FormControl>
-                    <Input {...field} />
-                  </FormControl>
-                </FormItem>
-              )}
-            />
-            <DialogFooter>
-              <Button
-                type="submit"
-                disabled={
-                  form.formState.isSubmitting || !form.formState.isDirty
-                }
-              >
-                Save
-              </Button>
-            </DialogFooter>
-          </form>
-        </Form>
-      </DialogContent>
-    </Dialog>
-  );
-};
-
 export const SystemActionsDropdown: React.FC<SystemActionsDropdownProps> = ({
   system,
   children,
 }) => {
-  const [open, setOpen] = useState(false);
   return (
-    <DropdownMenu open={open} onOpenChange={setOpen}>
-      <DropdownMenuTrigger asChild>{children}</DropdownMenuTrigger>
+    <DropdownMenu>
+      <DropdownMenuTrigger>{children}</DropdownMenuTrigger>
       <DropdownMenuContent>
-        <EditSystemDialog system={system} onSubmit={() => setOpen(false)}>
-          <DropdownMenuItem
-            onSelect={(e) => e.preventDefault()}
-            onClick={(e) => e.stopPropagation()}
-          >
-            Edit System
-          </DropdownMenuItem>
-        </EditSystemDialog>
+        <DropdownMenuGroup>
+          <EditSystemDialog system={system}>
+            <DropdownMenuItem
+              onSelect={(e) => e.preventDefault()}
+              onClick={(e) => e.stopPropagation()}
```

Comment:
in all the other dialogs I've never needed to apply both

Reply:
This is to prevent a sieffect from happening because the button in the EditSystemDialog is in a clickable element. (the row in the table). 

If we don't do this, it will instantly navigate to the system in that row. 

CC @adityachoudhari26

Reply:
Why is this pattern not used anywhere else in the code

Reply:
It was here before I touched this, and if I try and remove it—which I did many times—it will cause you to go to the `/[workspace]/systems/[system]/deployments` page. 

I think it is because it is in a clickable item already. 
<img width="1727" alt="Screenshot 2024-09-27 at 10 09 21 AM" src="https://github.com/user-attachments/assets/5846b8ff-77cd-4742-89a8-44ba47deebd9">

If you like I can try and change it so that works the way you expect.